app/models/concerns/orchestration/dns.rb

```diff
@@ -7,11 +7,15 @@ module Orchestration::DNS
   end
 
   def dns?
-    hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil? && host.managed? && managed?
+    # host.managed? and managed? should always come first so that orchestration doesn't
+    # even get tested for such objects
+    (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil?
```

Comment:
I think I see why you're checking for `host.nil?`, but the ip_available? and mac_available? methods don't look safe when `host` is nil.